From c7730bc66e254c87e092e4ef6baba769b56eee62 Mon Sep 17 00:00:00 2001
From: Radu-Andrei BULIE <radu.bulie@freescale.com>
Date: Thu, 17 May 2012 22:31:00 +0000
Subject: [PATCH 190/518] dpa_offload use case : Add support for L2 header
 size in the DPA IPSec code

Signed-off-by: Radu-Andrei BULIE <radu.bulie@freescale.com>
[Grabbed from the branch, LINUX_IR5.2.0, of
https://git.freescale.com/git-private/cgit.cgi/ppc/alu-b4860/linux.git.]
Signed-off-by: Tiejun Chen <tiejun.chen@windriver.com>
---
 drivers/staging/fsl_dpa_offload/usecases/xfrm_km.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/staging/fsl_dpa_offload/usecases/xfrm_km.c b/drivers/staging/fsl_dpa_offload/usecases/xfrm_km.c
index a5ddd1b..91d03b1 100644
--- a/drivers/staging/fsl_dpa_offload/usecases/xfrm_km.c
+++ b/drivers/staging/fsl_dpa_offload/usecases/xfrm_km.c
@@ -364,13 +364,13 @@ static int offload_sa(int dpa_ipsec_id,
 	sa_params->spi = x->id.spi;
 	sa_params->sa_bpid = usr_params.sa_bpid;
 	sa_params->crypto_params.alg_suite = alg_suite(x);
+	sa_params->l2_hdr_size = ETH_HLEN;
 	if (sa_params->crypto_params.alg_suite < 0)
 		return -ENOTSUPP;
 
 	if (dir == XFRM_POLICY_OUT) {
 		sa_params->sa_dir = DPA_IPSEC_OUTBOUND;
 		sa_params->start_seq_num = 1;
-		sa_params->l2_hdr_size = ETH_HLEN;
 		sa_params->sa_wqid = usr_params.sa_wqid;
 		if (x->props.family == AF_INET) {
 			memset(&outer_iphdr, 0, sizeof(outer_iphdr));
-- 
1.7.5.4

